third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/TensorMemoryToLLVM.cpp

```diff
@@ -256,6 +256,46 @@ static void lowerStoreToTensorMemory(Location loc, ModuleOp mod, Value src,
   b.barrier();
 }
 
+struct TensorMemoryObject {
+  TensorMemoryObject(Value base, int64_t rank, Location loc,
+                     RewriterBase &rewriter)
+      : base(base) {
+    auto b = TritonLLVMOpBuilder(loc, rewriter);
+    offsets.append(rank, b.i32_val(0));
```

Comment:
do we need the offsets? Looks like they are always 0?

Reply:
yeah, first I tried the trivial definition

```
struct TensorMemoryObject {
   Value base;
};
```
but got an LLVM error `error: LLVM Translation failed for operation: builtin.unrealized_conversion_cast`. Maybe I need to modify the type converter you showed above

Reply:
yes you need to modify the place I pointed out otherwise there would be a type mismatch. You can check the memory space of the memdesc in the type convert and for tmem memdesc just return a i32 type.

Reply:
yes it works now, thanks. Now it's trivial